### DISCUSSION TITLE
Microcks gRPC mocks

### DISCUSSION THREAD

**Discussion by hguerrero** [microcks/microcks#1046] [Q&A] [answered]
When mocking gRPC, Microcks lets you add mocks with postman collections.

1. Do we need to always add the `.metadata` file? It looks like it works without, but it always returns the same value, no matter the payload.
2. Considering that you don't send the expected payload, what is the algorithm to return the mock? Is it the first item on the list? Or how does Microcks determine what to return?

**Comment by lbroudoux** [comment, marked as answer]
Hey @hguerrero!

1) Metadata file is no longer mandatory for a functional gRPC mock as we now have implemented https://github.com/microcks/microcks/issues/955 (support for empty / null dispatcher for gRPC).

2) Without dispatcher being specified, the mock just return the first item (response) in the list
